Mathlib/SetTheory/Game/Basic.lean

```diff
@@ -389,6 +399,50 @@ theorem quot_mul_comm (x y : PGame.{u}) : (⟦x * y⟧ : Game) = ⟦y * x⟧ :=
 theorem mul_comm_equiv (x y : PGame) : x * y ≈ y * x :=
   Quotient.exact <| quot_mul_comm _ _
 
+instance isEmpty_leftMoves_mul (x y : PGame.{u}) [IsEmpty x.LeftMoves] [IsEmpty y.RightMoves] :
+    IsEmpty (x * y).LeftMoves := by
```

Comment:
```suggestion
instance isEmpty_leftMoves_mul (x y : PGame.{u})
    [IsEmpty (x.LeftMoves × y.RightMoves ⊕ x.RightMoves × y.LeftMoves] :
    IsEmpty (x * y).LeftMoves := ‹_›
```
seems more principled (and avoids the instance duplication, and forgetting some combinations of instances as you have done now)

Reply:
The combinations I omitted correspond to when either game is identical to the zero game. So actually, your approach should generalize those instances too.

Reply:
Yes, that is indeed my point